components/forms/Form/Form.tsx

```diff
@@ -28,6 +35,47 @@ const InnerForm = (props: InnerFormProps & FormikProps<FormValues>) => {
   const serverErrorId = `${errorId}-server`;
   const formStatusError = props.status === "Error" ? t("server-error") : null;
 
+  const SITE_KEY = process.env.NEXT_PUBLIC_RECAPTCHA_V3_SITE_KEY;
+  const isReCaptchaEnableOnSite = process.env.RECAPTCHA_ENABLE === "true";
+
+  const handleSubmitWithRecaptcha = (e) => {
+    // avoid multiple re-call while on submit state.
+    if (isSubmitting) return false;
+    logMessage.debug(`reCaptcha On :  ${isReCaptchaEnableOnSite}`);
+    e.preventDefault();
+    window.grecaptcha.ready(() => {
+      // get reCAPTCHA response
+      window.grecaptcha.execute(SITE_KEY, { action: "submit" }).then(async (token) => {
+        await sendRecaptchaToken(token).then((res) => {
+          const recaptchaRes = res?.data as ReCaptchaResponse;
+          logMessage.warn(recaptchaRes);
+          // assuming you're not a Robot
+          // continue submission process
+          handleSubmit(e);
+        });
+      });
+    });
+  };
+
+  const sendRecaptchaToken = async (token: string) => {
+    // call a backend API to verify reCAPTCHA response
+    try {
+      return await axios({
+        url: "/api/verify",
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        data: {
+          userToken: token,
+        },
+        timeout: process.env.NODE_ENV === "production" ? 60000 : 80000,
+      });
+    } catch (error) {
+      logMessage.error(error);
```

Comment:
What happens if there is an error? It seems to me that the calling code on line 49 could end up with some poor data.